modules/apps/app-builder/app-builder-test/src/testFunctional/macros/FormViewBuilder.macro

```diff
@@ -22,11 +22,24 @@ definition {
 	}
 
 	macro addFieldByDoubleClickFromObjectSidebar {
-		var key_fieldLabel = "${fieldLabel}";
+		var key_fieldLabel = "${fieldType}";
+
+		var iterator = "0";
+		var maxRetryNumber = "3";
+
+		while ((IsElementNotPresent(locator1 = "FormViewBuilder#SIDEBAR_BASIC_TAB")) && ("${iterator}" != "${maxRetryNumber}")) {
+			DoubleClick(locator1 = "FormViewBuilder#OBJECT_FIELD_LABEL");
+
+			Pause(locator1 = "1000");
+
+			if ("${iterator}" == "${maxRetryNumber}") {
+				echo("Couldn't add the field by double-click action, maximum retries number reached: ${iterator}");
+			}
 
-		DoubleClick(locator1 = "FormViewBuilder#OBJECT_FIELD_LABEL");
+			var iterator = MathUtil.sum("${iterator}", "1");
+		}
 
-		FormViewBuilder.validateFieldLabel(fieldLabel = "${fieldLabel}");
+		AssertElementPresent(locator1 = "FormViewBuilder#SIDEBAR_BASIC_TAB");
```

Comment:
You can remove it. Asserts should be added only inside the test or inside macros that is meant to assert something.